alibi_detect/cd/tensorflow/preprocess.py

```diff
@@ -6,34 +6,57 @@
 from alibi_detect.utils.tensorflow.prediction import predict_batch, predict_batch_transformer
 
 
+class _Encoder(tf.keras.Model):
+    def __init__(
+            self,
+            input_layer: Union[tf.keras.layers.Layer, tf.keras.Sequential, tf.keras.Model],
```

Comment:
Technically every `tf.keras.Sequential` model is an instance of `tf.keras.Model`, so we could simplify all type hints to just `tf.keras.Model`? Hopefully people wouldn't take it as `tf.keras.Sequential` is not allowed.

Reply:
Good point, I will raise this in a separate issue since both are used across the library.